sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/ServiceItemLease.java

```diff
@@ -236,4 +234,32 @@ public String toString() {
             this.getTimestamp(),
             UNIX_START_TIME.plusSeconds(Long.parseLong(this.getTs())));
     }
+
+    @SuppressWarnings("serial")
+    static final class ServiceItemLeaseJsonSerializer extends StdSerializer<ServiceItemLease> {
```

Comment:
we should add a deserializer as well. Document/CosmosItemProperties can use the custom deserializer.

You can take a look at `PartitionKeyInternal.java` as example.

Reply:
Won't fix.

Reply:
> Won't fix.

why?

Reply:
already explained "why"; you're asking to add dead code because there's no direct Json deserialization that is needed for ServiceItemLease at this time. The "deserialization" is done via fromDocument() method call.

Reply:
the explanation on fromDocument() makes sense. so that acts as the deserializer, that's what needed. thanks.